orttraining/orttraining/training_api/module.cc

```diff
@@ -202,12 +214,13 @@ Status Module::TrainStep(const std::vector<OrtValue>& inputs, std::vector<OrtVal
   feeds.insert(feeds.end(), weights_.begin(), weights_.end());
   feeds.insert(feeds.end(), gradients_.begin(), gradients_.end());
   // TODO: consider maintaining this as ortvalue instead of bool
-  OrtValue do_update_input;
-  utils::WrapInOrtValue<bool>(accumulate_gradient_, &do_update_input);
-  feeds.push_back(do_update_input);
+  OrtValue reset_grad_input;
+  utils::WrapInOrtValue<bool>(!accumulate_gradient_, &reset_grad_input);
+  feeds.push_back(reset_grad_input);
 
-  // TODO: need to filter out the grads from the output ortvalues
+  outputs.resize(train_output_names_.size());
   auto status = train_sess_->Run(RunOptions(), train_input_names_, feeds, train_output_names_, &outputs);
+  outputs.erase(outputs.begin()+train_user_output_size_, outputs.end());
```

Comment:
user output is far more less then internal outputs (e.g. grads), So would it be better we picked up thos user ouput and put it into final results, instead of erasing more internal outputs.

Reply:
Alternatively, since we know the user output names, can we invoke `Run` with only those user output names? Then we don't need to worry ourselves with the freeing that memory either.

Reply:
The reason why we have those graph outputs are to trigger the gradient update, so I suspect that the accumulation nodes will not be executed if we do this. I will test it in any case

Reply:
Okay after testing, the behaviour of inference session is such that it will still generate all graph outputs but we can give it partial fetch names, so we can do what you suggested @baijumeswani , thanks for the suggestion!